communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,246 @@
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class PomAnalyzer {
+
+    @Inject
+    private PomReader pomReader;
+
+    @Inject
+    @DACartographerCore
+    private CartographerCore carto;
+
+    public GAVDependencyTree readRelationships(File pomRepoDir, File pomPath)
+            throws PomAnalysisException {
+
+        try {
+            GAVDependencyTree root = null;
+
+            File tempDir = Files.createTempDirectory("deps").toFile();
+
+            Map<GAV, GAVDependencyTree> gavDependencyTreeMap = new HashMap<>();
+
+            try {
+                // map of file and ProjectVersionRefs that represent all the pom.xmls found in the pomRepoDir directory
+                Map<File, ProjectVersionRef> projectVersionRefs = getProjectVersionRefs(tempDir,
+                        findAllPomFiles(pomRepoDir));
+
+                // for each pom.xml
+                for (File path : projectVersionRefs.keySet()) {
+                    ProjectVersionRef ref = projectVersionRefs.get(path);
+
+                    // find all the dependencies / plugins / parents / plugins etc for each pom.xml
+                    Set<ProjectRelationship<?, ?>> relationships = getDiscoveryResult(tempDir,
+                            pomRepoDir, ref).getAcceptedRelationships();
+
+                    boolean rootRelationship = false;
+
+                    // need to use canonical path to get better way of knowing if 2 paths are the same
+                    if (pomPath.getCanonicalPath().equals(path.getCanonicalPath())) {
+                        rootRelationship = true;
+                    }
+
+                    for (ProjectRelationship<?, ?> relationship : relationships) {
+                        ProjectVersionRef origin = relationship.getDeclaring();
+                        ProjectVersionRef target = relationship.getTarget();
+
+                        // convert from ProjectVersionRef to GAV
+                        GAV originGAV = generateGAV(origin);
+                        GAV targetGAV = generateGAV(target);
+
+                        // get the origin GAVDependencyTree
+                        GAVDependencyTree originScmGAVTree = addGAVDependencyTreeToMapper(
+                                gavDependencyTreeMap, originGAV);
+                        GAVDependencyTree targetScmGavTree = addGAVDependencyTreeToMapper(
+                                gavDependencyTreeMap, targetGAV);
+
+                        if (rootRelationship)
+                            root = originScmGAVTree;
```

Comment:
Yeah I could just add a boolean to only set the origin once, then be done with it, but I kinda think it's more readable the current way. However you raised a good point; this might look confusing to readers. Maybe I'll add a comment somewhere.